src/abstractdataframe/show.jl

```diff
@@ -405,14 +405,28 @@ function showrows(io::IO,
     end
 
     rowmaxwidth = maxwidths[ncols + 1]
+    colwidths = maxwidths .+ length(" ") .+ length(" |")  # add border + padding
+    colwidths[end] += length("|")  # add row column initial border
     chunkbounds = getchunkbounds(maxwidths, splitcols, displaysize(io)[2])
     nchunks = allcols ? length(chunkbounds) - 1 : min(length(chunkbounds) - 1, 1)
 
+    overflowsdisplay = false
     header = displaysummary ? summary(df) : ""
+
+    # when allcols=true and any column overflows display, print omission message
+    if allcols && any(colwidths[end] .+ colwidths[1:end-1] .> displaysize(io)[2])
```

Comment:
when `allcols=true` we should disregard displaysize and just print everything if `splitcols=false`. What you propose should be done only if `splitcols=true` (`splitcols=false` means that the user wants to force printing everything as it is).